airbyte-integrations/connectors/source-mintegral/source_mintegral/streams.py

```diff
@@ -61,3 +53,97 @@ class Campaigns(MintegralStream):
 
     def path(self, **kwargs) -> str:
         return "campaign"
+
+
+class MintegralReportingStream(HttpStream, IncrementalMixin):
+    page_size = 500
+    backfill_days = 2
+    url_base = "https://ss-api.mintegral.com/api/v1/reports/data"
+
+    def __init__(self, authenticator: TokenAuthenticator, **kwargs):
+        self._state = {}
+        super().__init__(authenticator=authenticator)
+
+    def log(self, message):
+        print(message)  # oddly enough logger.debug() os not printed in airbyte logs, but prints are
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return "date"
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        self._state[self.cursor_field] = value[self.cursor_field]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Dict[str, Any]]:
+        response_json = response.json()
+        current_page = int(response_json["page"])
+        if current_page < int(response_json["page_count"]):
+            return {"page": current_page + 1}
+        return None
+
+    def request_params(
+            self,
+            stream_state: Optional[Mapping[str, Any]],
+            stream_slice: Optional[Mapping[str, Any]] = None,
+            next_page_token: Optional[Mapping[str, Any]] = None,
+    ):
+        request_params = {
+            "start_date": stream_slice[self.cursor_field],
+            "end_date": stream_slice[self.cursor_field],
+            "per_page": self.page_size,
+            "page": next_page_token["page"] if next_page_token else 1
+        }
+        self.log(f"Request params: {request_params}")
+        return request_params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Dict]:
+        if response.text:
+            response_json = response.json()
+            self.log(f"Page {response_json['page']}, per_page {response_json['per_page']}, "
+                     f"page_count {response_json['page_count']}, total_count {response_json['total_count']}")
+            yield from response_json["data"]
+        else:
+            yield {}
+
+    def read_records(
```

Comment:
Is it me or this method is identical to the one it overrides?

Reply:
It is, I forgot to remove it during the dev process as it did contain some state managing logic at some point